File: frontends/poly-restaurant/src/app/models/recipe.ts
export interface Recipe {
    shortName: string;
    post: Recipe.PostEnum;
    cookingSteps: Array<string>;
    meanCookingTimeInSec: number;
}
export namespace Recipe {
    export type PostEnum = 'BAR' | 'COLD_DISH' | 'HOT_DISH';
    export const PostEnum = {
        BAR: 'BAR' as PostEnum,
        COLDDISH: 'COLD_DISH' as PostEnum,
        HOTDISH: 'HOT_DISH' as PostEnum
    };
}

File: bff-backend/src/main/java/fr/univcotedazur/si5ci/microrestaurant/teamf/bffbackend/models/kitchen/Recipe.java
package fr.univcotedazur.si5ci.microrestaurant.teamf.bffbackend.models.kitchen;

import lombok.Data;

import java.io.Serializable;
import java.util.List;

@Data
public class Recipe implements Serializable {
    private String shortName;
    private Post post;
    private List<String> cookingSteps;
    private int meanCookingTimeInSec;
}
